debezium-core/src/main/java/io/debezium/transforms/outbox/EventRouter.java

```diff
@@ -230,10 +244,15 @@ private Schema getValueSchema(Schema debeziumEventSchema, Integer version) {
     private SchemaBuilder getSchemaBuilder(Schema debeziumEventSchema) {
         SchemaBuilder schemaBuilder = SchemaBuilder.struct();
 
+        Schema deletedSchema = (debeziumEventSchema.field(fieldIsDeleted) == null)
+                ? SchemaBuilder.bool().optional()
+                : debeziumEventSchema.field(fieldIsDeleted).schema();
+
         // Add default fields
         schemaBuilder
                 .field(ENVELOPE_EVENT_TYPE, debeziumEventSchema.field(fieldEventType).schema())
-                .field(ENVELOPE_PAYLOAD, debeziumEventSchema.field(fieldPayload).schema());
+                .field(ENVELOPE_PAYLOAD, debeziumEventSchema.field(fieldPayload).schema())
+                .field(ENVELOPE_DELETED, deletedSchema);
```

Comment:
For propagating this field, is there any explict logic even needed, couldn't this be even already done right now using the "additional fields" logic?

Reply:
Awesome. Looking forward to seeing this being finalized, it's a great improvement to the outbox support.

Reply:
Had some work this come up but hopping get some work done this evening.

Reply:
Cool; thanks for the heads-up!